prompt2model/dataset_retriever/dataset_index_file/preprocessing.py

```diff
@@ -0,0 +1,69 @@
+"""Filtering out datasets before we do heavy processing on them."""
+import json
+from typing import Any
+
+from huggingface_hub import list_datasets
+
+# Constants
```

Comment:
I don't know if this needs to be fixed right now, but just a note that it would probably be better to do this through argparse. If you're using github copilot it will happily write up the argparse code for you :)